zeus/optimizer/batch_size/common.py

```diff
@@ -85,32 +84,75 @@ def _check_default_batch_size(cls, values: Dict[str, Any]) -> Dict[str, Any]:
         return values
 
 
-class JobConfig(JobSpec):
+class JobSpec(JobParams):
+    """Job specification that user inputs.
+
+    Attributes:
+        job_id: ID of job
+        job_id_prefix: Prefix of job.
```

Comment:
Are we printing that ID to stdou? I think having prefix & id both might be a bit confusing cuz they have to pick it up, save it, and set up the env variable. Why can't we just use prefix as an ID?

Reply:
I see. Why the users cannot designate globally unique IDs though? Is it because of the interface of finding out the global id is submitting a job? 

My thought was, since `uuid` is really hard to set by a user (due to required format), the user can just put whatever `job_id` that is easier for them to read/write to make a globally unique ID, just like we set up an user ID.

Reply:
Sorry, I lost track of this. You would have experienced this: When you try to designate your user ID in some service when you sign up, sometimes it says it's already taken. This is because the user does not have a complete view of all the IDs registered in the service and fundamentally find an ID that does not already exist. So the solution for online services is to make the user retry indefinitely until they hit a user ID that does not already exist. This solution does not work for the BSO, because, if the user accidentally specifies a job ID that already exists, that training run will be treated as a trial of that job, merging together unrelated training runs. Thus, we need the server to eventually give a new job a globally unique ID that does not overlap with any existing job. However, if we just use `uuid4`, as you mentioned, it's very hard for the user to handle or distinguish. So, the role of `job_id_prefix` is to allow the user to include a string that makes it easy for them to distinguish/identify jobs. Then, the server appends a git-commit-like hash to `job_id_prefix` to make it unique.

Reply:
Ahhh I see. Didn't think about the case of accidental trials.